### DISCUSSION TITLE
appDir: what wrong with prefetch behavior?

### DISCUSSION THREAD

**Discussion by ji2k** [vercel/next.js#43408] [Help]
![image](https://user-images.githubusercontent.com/61374202/204097568-f2343ae1-2ac3-4aea-849c-8ab66735c4cd.png)
Always have to wait for a second request when navigating to another route, even though it's already prefetched.

**Comment by rshaul** [comment]
I'm also seeing problems where I load the root index page and it is prefetching random pages that are not even linked to anywhere in the site. Or are linked to several navigations deeper. Setting prefetch=false on the Links completely disables client side routing so that isn't a great work around.
The extra client side JS downloads is making the app dir have worse lighthouse scores than the pages version.

Edit: The random prefetching was caused by context, for some reason the pages that were using the context provided in the root layout was triggering prefetching of those pages. Just replaced context with RSC props passing and it's better.
